slack_ros/scripts/slack_ros.py

```diff
@@ -34,46 +34,72 @@ def __init__(self):
         self.username = rospy.get_param('~username', 'ros-bot')
 	
         # Create a publisher for our custom message.
-        pub = rospy.Publisher('from_slack_to_ros', String, queue_size=10)
-	rospy.Subscriber("from_ros_to_slack", String, self.callback)
-	rospy.Subscriber("send_file_to_slack", String, self.filecallback)
+        self.pub = rospy.Publisher('from_slack_to_ros', String, queue_size=10)
+        rospy.Subscriber("from_ros_to_slack", String, self.message_callback)
+        rospy.Subscriber("send_file_to_slack", String, self.file_callback)
 
-	# Create the slack client
-	self.sc = SlackClient(self.token)
+        # Create the slack client
+        self.slack_client = SlackClient(self.token)
+        self.has_connection = False
+        self.subscribed_msg_store = []
```

Comment:
My initial idea was exactly the same as it has been described above, but I could not find a way to block the callback. One way is to unsubscribe/resubscribe and that's not really blocking the callback, and again whether that is efficient is questionable. Or use semaphore inside the callback method itself, I have not tried this and don't know if threads would go to sleep while blocking the callback.

Any idea is welcome, meanwhile I would further look into this and update the PR.